### PR TITLE
Updated the user openapi definition with a real model.

### PR DESCRIPTION
**Pull request by tzfx**
- Updated our user model documentation with an openapi3 compliant schema that reflects reality.
- Documented the Team model.
- Modified the swagger-jsdoc configuration to allow us to keep the jsdoc in with the *.model.js file.

### CODE REVIEW

File: src/app/core/user/user.model.js [hunk 1]
```diff
@@ -304,6 +375,10 @@ UserSchema.statics.fullCopy = function(user) {
 		if (_.has(toReturn, 'password')) {
 			delete toReturn.password;
 		}
+		if (_.has(toReturn, 'salt')) {
```
Comment (tzfx) [inline src/app/core/user/user.model.js:378]: I assume we didn't actually want to return the salt property, even if it's behind an admin role auth.

File: src/lib/express.js [hunk 2]
```diff
@@ -222,7 +222,10 @@ function initSwaggerAPI(app) {
 			},
 			basePath: baseApiPath
 		},
-		apis: config.files.routes.map((route) => path.posix.resolve(route))
+		apis: [
+			'./src/app/**/*.routes.js',
+			'./src/app/**/*.model.js'
+		]
```
Comment (jrassa) [inline src/lib/express.js]: This removes the resolution from relative to absolute paths.  Not sure how necessary it is, but may effect production deployments.  Also the paths to for routes and models are currently configured in `config/assets.js`.  This duplicates that config to multiple places.  Updating the config to the following should address both of these issues.
```
apis: [
    ...config.files.routes.map((route) => path.posix.resolve(route)),
    ...config.files.model.map((model) => path.posix.resolve(model))
]
```

File: src/app/core/user/user.model.js (hunk 1 above)
Comment (jrassa) [reply src/app/core/user/user.model.js:378]: Good catch.

File: src/lib/express.js (hunk 2 above)
Comment (tzfx) [reply src/lib/express.js]: Good plan!

**Pull request merged**